src/platform/packages/private/kbn-grid-layout/grid/grid_row/grid_row_drag_preview.tsx

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React, { useEffect } from 'react';
+
+import { UseEuiTheme } from '@elastic/eui';
+import { css } from '@emotion/react';
+
+import { useGridLayoutContext } from '../use_grid_layout_context';
+
+export const GridRowDragPreview = React.memo(({ rowId }: { rowId: string }) => {
+  const { gridLayoutStateManager } = useGridLayoutContext();
+
+  useEffect(
+    () => {
+      return () => {
+        // when drag preview unmounts, this means the header was dropped - so, scroll to it
+        const headerRef = gridLayoutStateManager.headerRefs.current[rowId];
+        headerRef?.scrollIntoView({ behavior: 'smooth', block: 'center' });
+      };
```

Comment:
I'm not sure if we want this type of behaviour in the `kbn-grid-layout` package? But it's easiest to put it here rather than doing some wild `gridLayout` diffing to figure out which grid element moved 🙈 We could always make this some sort of `onRowMoved` callback if we wanted to make it more generic as a follow up - interested in thoughts

Reply:
I think this behaviour is completely fine for now!